[CI] IndicesClientIT.testDataStreams failure

**Build scan**: https://gradle-enterprise.elastic.co/s/rtz2uf3u5fuwm

**Repro line**: ./gradlew ':client:rest-high-level:asyncIntegTestRunner' --tests "org.elasticsearch.client.IndicesClientIT.testDataStreams" -Dtests.seed=E89C70116AD44583 -Dtests.security.manager=true -Dtests.locale=ja-JP-u-ca-japanese-x-lvariant-JP -Dtests.timezone=Australia/LHI -Druntime.java=8

**Reproduces locally?**: No

**Applicable branches**: 7.9

**Failure history**: First time

**Failure excerpt**: 

```
org.elasticsearch.client.IndicesClientIT > testDataStreams FAILED
    java.lang.AssertionError: 
    Expected: not <0L>
         but: was <0L>
        at __randomizedtesting.SeedInfo.seed([E89C70116AD44583:AE38E9FD1394286D]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.client.IndicesClientIT.testDataStreams(IndicesClientIT.java:2084)
```

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java
@@ -1590,6 +1590,9 @@ public class IndicesClientIT extends ESRestHighLevelClientTestCase {
         IndicesClient indices = highLevelClient().indices();
         response = execute(createDataStreamRequest, indices::createDataStream, indices::createDataStreamAsync);
         assertThat(response.isAcknowledged(), equalTo(true));
+        ensureHealth(dataStreamName, (request -> {
+            request.addParameter("wait_for_status", "yellow");
+        }));
 
         GetDataStreamRequest getDataStreamRequest = new GetDataStreamRequest(dataStreamName);
         GetDataStreamResponse getDataStreamResponse = execute(getDataStreamRequest, indices::getDataStream, indices::getDataStreamAsync);

